lib/src/facts/windows/identity_resolver.cc

```diff
@@ -20,20 +20,37 @@ namespace facter { namespace facts { namespace windows {
         auto nameformat = NameSamCompatible;
         GetUserNameExW(nameformat, nullptr, &size);
         if (GetLastError() != ERROR_MORE_DATA) {
-            LOG_DEBUG("failure resolving identity facts: %1% (%2%)", leatherman::windows::system_error());
+            LOG_DEBUG("failure resolving identity facts: %1% (%2%)", leatherman::windows::system_error(), GetLastError());
             return result;
         }
 
         // Use the string as a raw buffer that supports move and ref operations.
         wstring buffer(size, '\0');
         if (!GetUserNameExW(nameformat, &buffer[0], &size)) {
-            LOG_DEBUG("failure resolving identity facts: %1% (%2%)", leatherman::windows::system_error());
+            LOG_DEBUG("failure resolving identity facts: %1% (%2%)", leatherman::windows::system_error(), GetLastError());
             return result;
         }
 
         // Resize the buffer to the returned string size.
         buffer.resize(size);
         result.user_name = boost::nowide::narrow(buffer);
+
+        // Build the well known local Administrators group SID
+        PSID local_administrators_sid;
```

Comment:
There's an is_admin helper in Leatherman. I'm pretty sure it does the same thing. https://github.com/puppetlabs/leatherman/blob/master/windows/src/user.cc#L14-L45